vespa-cloud/vespa-documentation-search/src/main/java/ai/vespa/cloud/docsearch/OutLinksDocumentProcessor.java

```diff
@@ -0,0 +1,122 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package ai.vespa.cloud.docsearch;
+
+import com.yahoo.docproc.DocumentProcessor;
+import com.yahoo.docproc.Processing;
+import com.yahoo.document.DataType;
+import com.yahoo.document.Document;
+import com.yahoo.document.DocumentOperation;
+import com.yahoo.document.DocumentPut;
+import com.yahoo.document.DocumentType;
+import com.yahoo.document.DocumentUpdate;
+import com.yahoo.document.datatypes.Array;
+import com.yahoo.document.datatypes.StringFieldValue;
+import com.yahoo.document.datatypes.WeightedSet;
+import com.yahoo.document.update.FieldUpdate;
+import com.yahoo.documentapi.AsyncParameters;
+import com.yahoo.documentapi.AsyncSession;
+import com.yahoo.documentapi.Response;
+import com.yahoo.documentapi.ResponseHandler;
+import com.yahoo.documentapi.Result;
+import com.yahoo.documentapi.messagebus.MessageBusDocumentAccess;
+
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.Set;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+/**
+ * Extract unique file-links from this document, and add as an in-link in the target document.
+ * Use async Document API to set in-links, fire-and-forget.
+ * This is an example Document Processor / use of Document API - and to be tested in a SystemTest.
+ */
+public class OutLinksDocumentProcessor extends DocumentProcessor {
+    private static final Logger logger = Logger.getLogger(OutLinksDocumentProcessor.class.getName());
+
+    static final String DOC_DOCUMENT_TYPE   = "doc";
+    static final String PATH_FIELD_NAME     = "path";
+    static final String OUTLINKS_FIELD_NAME = "outlinks";
+    static final String INLINKS_FIELD_NAME  = "inlinks";
+
+    private final MessageBusDocumentAccess access = new MessageBusDocumentAccess();
```

Comment:
Shouldn't this be an injected `DocumentAccess`? :D

Reply:
yes - but public is not yet on 7.291 so must add later

Reply:
What? 7.291 is already open-sourced, though ... >_<

Reply:
That is, released to maven central.